proxy/src/coco/peer.rs

```diff
@@ -307,6 +307,34 @@ impl Api {
         Ok(meta)
     }
 
+    /// Returns a feed of untracked [`project::Project`]s
+    ///
+    /// # Errors
+    ///
+    ///   * Retrieving the feed fails
+    #[allow(clippy::unused_self)]
+    pub fn discover_projects(&self) -> Result<Vec<DiscoveryItem>, error::Error> {
```

Comment:
```suggestion
    pub fn discover_projects() -> Result<Vec<DiscoveryItem>, error::Error> {
```
Instead of introducing this exception we can convert this method into an [associated function](https://doc.rust-lang.org/reference/items/associated-items.html).

Reply:
To the comment above about domain boundaries this method should live in the `project` module.